extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -624,6 +624,24 @@ void ConsumeWindowsEventLog::putEventRenderFlowFileToSession(const EventRender&
     const std::string& str_;
   };
 
+  DYNAMIC_TIME_ZONE_INFORMATION tzinfo;
+  auto ret = GetDynamicTimeZoneInformation(&tzinfo);
+  std::wstring tzstr;
+  std::string tzbias;
+  bool dst = false;
+  switch (ret) {
+  case TIME_ZONE_ID_UNKNOWN:
+    logger_->log_error("Failed to get timezone information!");
+    break;
+  case TIME_ZONE_ID_DAYLIGHT:
+    tzstr = std::wstring(tzinfo.DaylightName, 32);
+    dst = true;
+  case TIME_ZONE_ID_STANDARD:
+    tzstr = tzstr.empty() ? std::wstring(tzinfo.StandardName, 32) : tzstr;  // Use standard timezome name in case there is no daylight name or in case it's not DST
+    tzbias = std::to_string(tzinfo.Bias + (dst ? tzinfo.DaylightBias : 0));
```

Comment:
It may just be a fancy alias for zero, but `DYNAMIC_TIME_ZONE_INFORMATION` has a `StandardBias` field.  So this could be written nicer (and possibly safer) as `dst ? tzinfo.DaylightBias : tzinfo.StandardBias`

Reply:
Agreed, good spot, thanks!